libp2p/Session.cpp

```diff
@@ -248,6 +255,24 @@ void Session::write()
             DEV_GUARDED(x_framing)
             {
                 m_writeQueue.pop_front();
+                if(m_writeQueue.empty() && disconnectImminent)
+                {
+                    ba::io_context ioc;
```

Comment:
I think you should re-use the Host's io_context (`Host::m_ioContext`) - one way to do this would be to create the timer in `Host::startPeerSession` and pass it into the Session ctor. @gumb0 - thoughts?

Reply:
Yes, timer should better use `io_context` of `Host`.

Not sure about passing timer to constructor. `Session` has a pointer to `Host`, maybe better to introduce a public method in `Host` that will create a timer? This way we won't create it until it's really needed in `disconnect`.

Reply:
Alright!!

Reply:
> 
> 
> Yes, timer should better use `io_context` of `Host`.
> 
> Not sure about passing timer to constructor. `Session` has a pointer to `Host`, maybe better to introduce a public method in `Host` that will create a timer? This way we won't create it until it's really needed in `disconnect`.

Good idea, this way we can also easily create timers in other classes if needed.